src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -288,7 +288,7 @@ const TopicFragment = ({
           data={data}
           renderItem={(props) => renderItem({...props, section: 'suggested-communities'})}
           keyExtractor={(item, index) => index.toString()}
-          onEndReached={() => fetchData()}
+          onEndReached={() => (firstData.length > 0 ? fetchData() : null)}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change made here is a good one. It adds a condition to check if `firstData` array has any elements before calling the `fetchData()` function. This prevents unnecessary calls to `fetchData()` when there's no data, which can improve performance and reduce unnecessary network requests.
```diff
- onEndReached={() => fetchData()}
+ onEndReached={() => (firstData.length > 0 ? fetchData() : null)}
```
However, it would be better to use a more explicit check for the length of the array. In JavaScript, `0` is falsy, so `firstData.length > 0` will return `false` if the length is `0`. But for readability and to avoid potential confusion, I suggest using `firstData.length !== 0` instead.
```diff
- onEndReached={() => (firstData.length > 0 ? fetchData() : null)}
+ onEndReached={() => (firstData.length !== 0 ? fetchData() : null)}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->